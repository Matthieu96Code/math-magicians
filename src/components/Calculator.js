import React, { useState } from 'react';
import calculate from '../logic/calculate';
import '../styles/calculator.css';

const Calculator = () => {
  const [result, setResult] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const btnClick = (e) => {
    const operation = calculate(result, e.target.textContent);
    setResult(operation);
  };
  return (
    <div className="calculator-page">
      <h2>Let&apos;s do some math!</h2>
      <div className="calculator">
        <div className="screen">{result.next || result.total || '0'}</div>
        <div className="buttons">
          <button onClick={btnClick} type="button" className="calc-btn btn-ac special">AC</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-+/- special">+/-</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-% special">%</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-รท operation">รท</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-7 numbers">7</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-8 numbers">8</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-9 numbers">9</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-x operation">x</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-4 numbers">4</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-5 numbers">5</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-6 numbers">6</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-- operation">-</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-1 numbers">1</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-2 numbers">2</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-3 numbers">3</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-+ operation">+</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-0 numbers">0</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-. numbers">.</button>
          <button onClick={btnClick} type="button" className="calc-btn btn-= operation">=</button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
