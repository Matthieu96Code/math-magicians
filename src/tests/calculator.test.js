import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const component = renderer.create(
    <Calculator />,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
