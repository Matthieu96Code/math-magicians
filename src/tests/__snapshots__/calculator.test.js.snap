// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator-page"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="screen"
    >
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="calc-btn btn-ac special"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calc-btn btn-+/- special"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calc-btn btn-% special"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calc-btn btn-รท operation"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="calc-btn btn-7 numbers"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calc-btn btn-8 numbers"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calc-btn btn-9 numbers"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calc-btn btn-x operation"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calc-btn btn-4 numbers"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calc-btn btn-5 numbers"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calc-btn btn-6 numbers"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calc-btn btn-- operation"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calc-btn btn-1 numbers"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calc-btn btn-2 numbers"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calc-btn btn-3 numbers"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calc-btn btn-+ operation"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="calc-btn btn-0 numbers"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calc-btn btn-. numbers"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calc-btn btn-= operation"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
